lib/ex_toolkit/naming.ex

```diff
@@ -140,6 +140,43 @@ defmodule ExToolkit.Naming do
     end
   end
 
+  @doc """
+  Extracts and capitalizes the first and last names and then shortens the first name from a given name and.
+
+  ## Parameters
+
+  - `name`: A string representing a full name, or `nil`.
+
+  ## Examples
+
+      iex> Naming.extract_shorten_first_last_name(nil)
+      ""
+
+      iex> Naming.extract_shorten_first_last_name("")
+      ""
+
+      iex> Naming.extract_shorten_first_last_name("john")
+      "John"
+
+      iex> Naming.extract_shorten_first_last_name("john doe smith")
+      "J. Smith"
+  """
+  @spec extract_shorten_first_last_name(nil | String.t()) :: String.t()
+  def extract_shorten_first_last_name(name) do
+    names =
+      name
+      |> String.trim()
+      |> String.split(" ")
+      |> Stream.filter(&String.match?(String.slice(&1, 0, 1), ~r/^\p{L}$/u))
+      |> Enum.map(&String.capitalize/1)
+
+    case names do
+      [] -> ""
+      [first] -> shorten_firstname(first)
+      [first | rest] -> first <> " " <> Enum.at(rest, -1)
+    end
```

Comment:
Why can't we just make it `name |> extract_first_last_name() |> shorten_first_name()`?

Reply:
Because I want 

```
> extract_shorten_first_last_name("John")
"John"
```

Instead of:

```
> extract_shorten_first_last_name("John")
"J."
```

Reply:
Then I think we should name it to extract_short_name/1 💡

And we should reuse the extract_first_last_name code and just do a case to then short the first name if it returns 2 names 💪

Reply:
Yes, it would required the String.split(" "). I don't think it is a big problem but to optimize and avoid that, we could just extract the common part into an aux function 💡 

```elixir
defp to_list_of_names(name) do
     name
      |> String.trim()
      |> String.split(" ")
      |> Stream.filter(&String.match?(String.slice(&1, 0, 1), ~r/^\p{L}$/u))
      |> Enum.map(&String.capitalize/1)
 end
```

Reply:
What do you see as down side?

Reply:
None really, after implementing the code looks clean too :laughing: